packages/docusaurus-theme-classic/src/theme/CodeBlock/index.tsx

```diff
@@ -244,11 +244,15 @@ export default ({
                   }
 
                   return (
-                    <div key={i} {...lineProps}>
-                      {line.map((token, key) => (
-                        <span key={key} {...getTokenProps({token, key})} />
-                      ))}
-                    </div>
+                    <pre className={styles.preTag}>
+                      <code className={styles.codeTag}>
+                        <div key={i} {...lineProps}>
```

Comment:
this is still not valid html

Reply:
And I think a comment is needed on why exactly `pre>code>span` is done here in the code, in both places, since this is not covered by tests, this is the only way to guarantee validity in the future.

Reply:
thanks, I'm not an expert but why is this not valid?

Reply:
@slorber https://github.com/facebook/docusaurus/pull/3178#discussion_r467114208

Reply:
@slorber I would still recommend setting up tests to check availability and validity. Any mistakes can be critical for those who care. 
In addition, it is often useful for correct indexing, the search on the site is of course important, but when the developer does not know something, he usually uses google to search, and internal search is the second step. 
It may be worth asking the community for help in this matter. And to do it so that it is really seen.